src/hotspot/cpu/x86/vm_version_x86.hpp

```diff
@@ -677,6 +677,7 @@ class VM_Version : public Abstract_VM_Version {
   static uint32_t cpu_stepping()          { return _cpuid_info.cpu_stepping(); }
   static int  cpu_family()        { return _cpu;}
   static bool is_P6()             { return cpu_family() >= 6; }
+  static bool is_P6_or_later()    { return cpu_family() == 6 || cpu_family() == 18 || cpu_family() == 19; }
```

Comment:
Do we need cpu_family() == 18  check ?
19 is for Diamond Rapids and 6 for all Xerons before it, including E-core variants.

Reply:
Removed cpu_family()==18.